src/lib/matcher.rs

```diff
@@ -407,3 +408,252 @@ impl MatcherFactory {
         }
     }
 }
+
+// Tests
+#[cfg(test)]
+pub mod tests {
+    use crate::matcher::*;
+    use bstr::ByteSlice;
+    use rstest::rstest;
+    use std::ops::Range;
+
+    // ############################################################################################
+    // Tests to_bitvec() fixed string
+    // ############################################################################################
+    #[rstest]
+    #[case("AG".as_bytes(), "AGG".as_bytes(), bitvec![1, 1, 0])] // fixed string with match
+    #[case("AT".as_bytes(), "CCC".as_bytes(), bitvec![0, 0, 0])] // fixed string with no match
+    #[case("AT".as_bytes(), "ATGAT".as_bytes(), bitvec![1, 1, 0, 1, 1])] // fixed string with multiple non-overlapping matches
+    #[case("AGAG".as_bytes(), "AGAGAGAG".as_bytes(), bitvec![1, 1, 1, 1, 1, 1, 1, 1])] // fixed string with overlapping matches
```

Comment:
fixme: remove `as_bytes()` and instead convert them below.  Ditto for all the other tests